powerapi-core/src/main/scala/org/powerapi/core/power/Power.scala

```diff
@@ -23,132 +23,165 @@
-  def apply(value: Double, unit: String)    = new RawPower(value, PowerUnitSystem(unit))
+  def lt(other: Power): Boolean = this < other
+
+  def lteq(other: Power): Boolean = this <= other
+
+  def minus(other: Power): Power = this - other
+
+  def mul(factor: Double): Power = this * factor
+
+  def plus(other: Power): Power = this + other
+}
+
+object RawPower {
 
   // limit on abs. value of powers in their units
   private final val max_mw = Double.MaxValue
   private final val max_w = max_mw / 1000.0
   private final val max_kw = max_w / 1000.0
   private final val max_Mw = max_kw / 1000.0
+
+  def apply(value: Double, unit: PowerUnit): Power = new RawPower(value, unit)
+
+  def apply(value: Double, unit: String): Power = new RawPower(value, PowerUnitSystem(unit))
+
+  implicit object RawPowerIsOrdered extends Ordering[RawPower] {
+    def compare(a: RawPower, b: RawPower): Int = a compare b
+  }
+
 }
 
 /**
- * Defines a power value.
- *
- * @author Loïc Huertas <l.huertas.pro@gmail.com>
- * @author Romain Rouvoy <romain.rouvoy@univ-lille1.fr>
- */
+  * Defines a power value.
+  *
+  * @author Loïc Huertas <l.huertas.pro@gmail.com>
+  * @author Romain Rouvoy <romain.rouvoy@univ-lille1.fr>
+  */
 final class RawPower(val value: Double, val unit: PowerUnit) extends Power {
   private val log = LogManager.getLogger
-  
-  private[this] def bounded(max: Double) = 0.0 <= value && value <= max
-  
+
+  def toWatts: Double = unit.toWatts(value)
+
   require(unit match {
-      case MILLIWATTS => bounded(max_mw)
-      case WATTS      => bounded(max_w)
-      case KILOWATTS  => bounded(max_kw)
-      case MEGAWATTS  => bounded(max_Mw)
-      case _ =>
-        val v = MEGAWATTS.convert(value, unit)
-        0.0 <= v && v <= max_Mw
-    }, "Power value is limited to 1.79e308 mW and cannot be negative")
-
-  def toMilliWatts = unit.toMilliWatts(value)
-  def toWatts      = unit.toWatts(value)
-  def toKiloWatts  = unit.toKiloWatts(value)
-  def toMegaWatts  = unit.toMegaWatts(value)
-  def toUnit(u: PowerUnit) = toMilliWatts / MILLIWATTS.convert(1, u)
-  
-  override def toString() = s"$value $unit"
-  
-  def compare(other: Power) = toMilliWatts compare other.toMilliWatts
-  
+    case MILLIWATTS => bounded(max_mw)
+    case WATTS => bounded(max_w)
+    case KILOWATTS => bounded(max_kw)
+    case MEGAWATTS => bounded(max_Mw)
+    case _ =>
+      val v = MEGAWATTS.convert(value, unit)
+      0.0 <= v && v <= max_Mw
+  }, "Power value is limited to 1.79e308 mW and cannot be negative")
+
+  def toKiloWatts: Double = unit.toKiloWatts(value)
+
+  def toMegaWatts: Double = unit.toMegaWatts(value)
+
+  def toUnit(u: PowerUnit): Double = toMilliWatts / MILLIWATTS.convert(1, u)
+
+  override def toString(): String = s"$value $unit"
+
+  def compare(other: Power): Int = toMilliWatts compare other.toMilliWatts
+
+  def toMilliWatts: Double = unit.toMilliWatts(value)
+
+  def +(other: Power): Power = add(other.value, other.unit)
```

Comment:
[Scalastyle] WARNING: Method name does not match the regular expression '^[a-z][A-Za-z0-9]*(_=)?$'